File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cs
public class Solution {
    public int[] TopKFrequent(int[] nums, int k) {
        if (k <= 0 || k > nums.Length)
            return [];

        var result = new int[k];
        var numsSpan = nums.AsSpan();
        var numCountsDic = new Dictionary<int, int>();

        foreach (var number in numsSpan)
        {
            if (numCountsDic.ContainsKey(number))
                numCountsDic[number]++;
            else
                numCountsDic.Add(number, 1);
        }

        result =
            [.. numCountsDic
            .OrderByDescending(static v => v.Value)
            .Take(k)
            .Select(p => p.Key)];

        return result;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        result = []
        for i in range(len(nums)):
            product = None
            for j in range(len(nums)):
                if i == j:
                    j += 1
                    continue
                if product == None:
                    product = nums[j]
                else:
                    product *= nums[j]
                
            result.append(product)
            i += 1
        
        return result
